Clustering/hybridEcalSplitter/src/DDStripSplitter.cc

```
@@ -0,0 +1,790 @@
+  }
+
+  if ( _stripAspectRatio<2.0 ) {
+    streamlog_out ( WARNING ) << " -- not a long enough strip, not worth splitting -- length, width, aspect ratio: " << 
+      _stripLength << " " << _stripWidth << " " << _stripAspectRatio << ", doing nothing" << endl;
+    return;
+  }
+
+  std::pair < TVector3, TVector3 > stripEnds;
+  std::vector <std::string> * toSplit;
+  //  std::vector <std::string> * stripSplitter;
+  int orientation;
+
+  std::map < IMPL::LCCollectionVec*, std::string > outputcolls;
+
+  if (_saveIntersections) {
+    intersectionHits = new IMPL::LCCollectionVec( LCIO::CALORIMETERHIT );
+    intersectionHits->setFlag(intersectionHits->getFlag()|( 1 << LCIO::RCHBIT_LONG)); // store position
+
+    stripEndsEvenCol = new IMPL::LCCollectionVec( LCIO::CALORIMETERHIT );
+    stripEndsEvenCol->setFlag(stripEndsEvenCol->getFlag()|( 1 << LCIO::RCHBIT_LONG)); // store position
+
+    stripEndsOddCol = new IMPL::LCCollectionVec( LCIO::CALORIMETERHIT );
+    stripEndsOddCol->setFlag(stripEndsOddCol->getFlag()|( 1 << LCIO::RCHBIT_LONG)); // store position
+  }
+
+  for (int icol=0; icol<2; icol++) { // loop over strip collections in even and odd layers (assumed to have perpendicular orientations)
+    switch (icol) {
+    case 0:
+      if ( _evenIsTransverse==1 ) {
+	orientation = TRANSVERSE;
+      } else if ( _evenIsTransverse==0 ) {
+	orientation = LONGITUDINAL;
+      } else {
+	cout << "ERROR: strip orientation undefined!!" << endl;
+	assert(0);
+      }
+      toSplit = &_ecalCollectionsEvenLayers;
+      break;
+    case 1:
+      if ( _evenIsTransverse==1 ) {
+	orientation = LONGITUDINAL;
+      } else if ( _evenIsTransverse==0 ) {
+	orientation = TRANSVERSE;
+      } else {
+	cout << "ERROR: strip orientation undefined!!" << endl;
+	assert(0);
+      }
+      toSplit = &_ecalCollectionsOddLayers;
+      break;
+    default:
+      streamlog_out ( ERROR ) << "ERROR crazy stuff!!! abandoning event..." << endl;
+      return;
```

Here also you should throw an exception, either
- `throw marlin::SkipEventException(this)`
- `throw marlin::StopProcessingException(this)`

---

done (also added in a couple of other places)